packages/api/endpoints/rules.js

```diff
@@ -60,26 +61,28 @@ async function get(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function post(req, res) {
+  const {
+    model = new models.Rule(),
+    dbClient = await getKnexClient(),
+  } = req.testContext || {};
+
+  const rule = req.body || {};
+  const name = rule.name;
+
+  if (await model.exists(name)) {
+    return res.boom.conflict(`A record already exists for ${name}`);
+  }
+
   try {
-    const data = req.body;
-    const name = data.name;
-
-    const model = new models.Rule();
-
-    try {
-      await model.get({ name });
-      return res.boom.conflict(`A record already exists for ${name}`);
-    } catch (error) {
-      if (error instanceof RecordDoesNotExist) {
-        const record = await model.create(data);
-
-        if (inTestMode()) {
-          await addToLocalES(record, indexRule);
-        }
-        return res.send({ message: 'Record saved', record });
-      }
-      throw error;
-    }
+    return await dbClient.transaction(async (trx) => {
+      const record = await model.create(rule);
+      const postgresRecord = await translateApiRuleToPostgresRule(record, dbClient);
+      await trx(tableNames.rules).insert(postgresRecord, 'cumulus_id');
```

Comment:
I think we need to reverse the order of operations here: do the Postgres insert first and then the Dynamo insert. The reason why is that right now, the Dynamo write could succeed and the Postgres write would fail, meaning the databases would get out of sync. But if you do the Postgres write first, then the Dynamo write is only reached if the Postgres write doesn't error, **and** if the Dynamo write errors, then the transactional Postgres write is rolled back.